src/engine/QueryExecutionTree.cpp

```diff
@@ -203,7 +208,9 @@ size_t QueryExecutionTree::getCostEstimate() {
 // _____________________________________________________________________________
 size_t QueryExecutionTree::getSizeEstimate() {
   if (_sizeEstimate == std::numeric_limits<size_t>::max()) {
-    if (_qec) {
+    if (_cachedResult) {
+      _sizeEstimate = _cachedResult->size();
```

Comment:
I think the `size()` might be bogus if `ResultTable._status != ResultTable::FINISHED`